blog/views.py

```diff
@@ -1,25 +1,55 @@
 from django.contrib import messages
-from django.shortcuts import get_object_or_404, render, redirect
+from django.shortcuts import get_object_or_404, render, redirect, HttpResponseRedirect
 
 from .models import Blog
 from .forms import BlogForm
 
 
 def blog_list(request):
-    pass
+    blog_list = Blog.objects.all()
+    context = {'blog_list': blog_list}
+    return render(request, 'blog/list.html', context)
 
 
-def blog_detail(request):
-    pass
+def blog_detail(request, pk):
+    context = {}
+    context["data"] = Blog.objects.get(id=pk)
+
+    return render(request, "blog/detail.html", context)
 
 
 def blog_new(request):
-    pass
+    context = {}
+
+    form = BlogForm(request.POST or None)
+    if form.is_valid():
+        form.save()
+
+    context['form'] = form
+
+    if request.method == "POST":
+        return HttpResponseRedirect("/")
+
+    return render(request, 'blog/create_view.html', context)
+
+
+def blog_edit(request, pk):
+    context = {}
+
+    obj = get_object_or_404(Blog, id=pk)
+    form = BlogForm(request.POST or None, instance=obj)
+
+    if form.is_valid():
+        form.save()
 
+    context["form"] = form
 
-def blog_edit(request):
-    pass
+    if request.method == "POST":
```

Comment:
move to before `form = BlogForm(request.POST`

Reply:
if i move to before `form = BlogForm(request.POST` changes are not saved on edit